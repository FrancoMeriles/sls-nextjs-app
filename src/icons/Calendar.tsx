import { Icon } from '@chakra-ui/react'

const Calendar = (props) => (
  <Icon viewBox="0 0 16 16" {...props}>
    <g fillOpacity=".8" fillRule="evenodd" fill="currentColor">
      <g fill="#000" fillRule="nonzero">
        <path
          d="M4.996 1.4l-.001 1h6.067v-1h1.2v1h2.337v12.2H1.4V2.4h2.395l.001-1h1.2zM13.399 6.8H2.6v6.6h10.799V6.8zm-9.604-3.2L2.6 3.602v1.998h10.799V3.6L12.262 3.6v1h-1.2v-1H4.995l.001 1h-1.2l-.001-1z"
          transform="translate(-1128 -464) translate(1108 158) translate(20 254) translate(0 52)"
        ></path>
      </g>
    </g>
  </Icon>
)

export default Calendar
