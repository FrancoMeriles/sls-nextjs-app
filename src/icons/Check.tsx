import { Icon } from '@chakra-ui/react'

const Check = (props) => (
  <Icon viewBox="0 0 32 32" {...props}>
    <path
      d="M16 8.85c1.593 0 3.064.521 4.253 1.402l-.962.91A5.851 5.851 0 1016 21.85a5.851 5.851 0 005.772-6.808l1.096-1.038A7.15 7.15 0 1116 8.85zm5.642 1.89l1.028 1.028-6.69 6.703-3.18-3.18 1.029-1.03 2.152 2.153 5.66-5.675z"
      fill="currentColor"
      fillRule="nonzero"
    />
  </Icon>
)

export default Check
