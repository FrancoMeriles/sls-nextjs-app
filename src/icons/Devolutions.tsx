import { Icon } from '@chakra-ui/react'

const Devolutions = (props) => (
  <Icon viewBox="0 0 14 12" {...props}>
    <path
      fill="currentColor"
      d="M2.474 7.2h7.225a2.7 2.7 0 1 0 0-5.4H7V.6h2.7a3.9 3.9 0 1 1 0 7.8H2.473l2.45 2.389-.839.859L.14 7.8l3.945-3.848.838.859L2.473 7.2z"
    ></path>
  </Icon>
)

export default Devolutions
