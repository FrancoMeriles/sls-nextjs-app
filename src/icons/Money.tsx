import { Icon } from '@chakra-ui/react'

const Money = (props) => (
  <Icon viewBox="0 0 20 20" {...props}>
    <path
      fill="currentColor"
      fillOpacity=".8"
      fillRule="nonzero"
      d="M10 1.002a9 9 0 1 1 0 18 9 9 0 0 1 0-18zm0 1.2a7.8 7.8 0 1 0 0 15.6 7.8 7.8 0 0 0 0-15.6zm.6 2.376l.001 1.323c.972.109 1.774.484 2.393 1.083l-.804 1.044c-.624-.624-1.476-.912-2.316-.912-.828 0-1.356.396-1.356.996 0 .624.816.828 1.764 1.068 1.32.336 2.892.744 2.892 2.484 0 1.219-.784 2.277-2.573 2.448L10.6 15.44H9.4v-1.343c-1.131-.133-1.98-.598-2.574-1.233l.78-1.08a3.453 3.453 0 0 0 2.532 1.104c1.152 0 1.608-.564 1.608-1.092 0-.72-.852-.936-1.824-1.188l-.294-.073c-1.233-.315-2.55-.76-2.55-2.327 0-1.165.906-2.096 2.323-2.299L9.4 4.578h1.2z"
    ></path>
  </Icon>
)

export default Money
