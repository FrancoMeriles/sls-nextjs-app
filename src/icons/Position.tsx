import { Icon } from '@chakra-ui/react'

const Position = (props) => (
  <Icon viewBox="0 0 14 19" {...props}>
    <path
      fill="currentColor"
      fillRule="evenodd"
      clipRule="evenodd"
      d="M13.3471 6.34707C13.3471 9.39028 11.3722 13.1849 7.45665 17.7792L7 18.315L6.54334 17.7792C2.62775 13.1849 0.652924 9.39028 0.652924 6.34707C0.652924 2.84168 3.4946 0 7 0C10.5054 0 13.3471 2.84168 13.3471 6.34707ZM9.41002 6.595C9.41002 7.92602 8.33102 9.00503 7 9.00503C5.66898 9.00503 4.58997 7.92602 4.58997 6.595C4.58997 5.26398 5.66898 4.18497 7 4.18497C8.33102 4.18497 9.41002 5.26398 9.41002 6.595Z"
    ></path>
  </Icon>
)

export default Position
