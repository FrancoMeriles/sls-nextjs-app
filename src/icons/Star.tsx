import { Icon } from '@chakra-ui/react'

const Star = (props) => {
  return (
    <>
      {props.paint === 'full' && (
        <Icon viewBox="0 0 10 10" {...props}>
          <path
            fill="currentColor"
            d="M5.056 8L1.931 9.648l.597-3.49L0 3.684l3.494-.509L5.056 0l1.562 3.176 3.494.51-2.528 2.471.597 3.491z"
          ></path>
        </Icon>
      )}
      {props.paint === 'half' && (
        <Icon viewBox="0 0 10 10" {...props}>
          <g fill="none" fillRule="evenodd">
            <path
              fill="#e0e0e0"
              d="M5.256 8L2.131 9.648l.597-3.49L.2 3.684l3.494-.509L5.256 0l1.562 3.176 3.494.51-2.528 2.471.597 3.491z"
            ></path>
            <path
              fill="currentColor"
              d="M5.272 8.026L2.137 9.679l.6-3.502L.2 3.697l3.505-.51L5.272 0z"
            ></path>
          </g>
        </Icon>
      )}
      {props.paint === 'empty' && (
        <Icon viewBox="0 0 10 10" {...props}>
          <path
            fill="#e0e0e0"
            d="M5.056 8L1.931 9.648l.597-3.49L0 3.684l3.494-.509L5.056 0l1.562 3.176 3.494.51-2.528 2.471.597 3.491z"
          ></path>
        </Icon>
      )}
    </>
  )
}

export default Star
