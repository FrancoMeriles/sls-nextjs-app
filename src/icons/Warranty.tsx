import { Icon } from '@chakra-ui/react'

const Warranty = (props) => (
  <Icon viewBox="0 0 12 14" {...props}>
    <path
      fill="currentColor"
      fillRule="nonzero"
      d="M5.305 10.354l-1.686 3.34-1.24-2.458H0l1.64-3.202a5.2 5.2 0 1 1 8.588.194l1.54 3.008H9.39l-1.241 2.458-1.673-3.315a5.266 5.266 0 0 1-1.171-.025zm-1.001-.237a5.201 5.201 0 0 1-1.975-1.234l-.693 1.353h1.357l.626 1.239.685-1.358zm3.192.064l.653 1.294.625-1.239h1.358l-.617-1.204a5.193 5.193 0 0 1-2.019 1.15zM6 9.2a4 4 0 1 0 0-8 4 4 0 0 0 0 8z"
    ></path>
  </Icon>
)

export default Warranty
