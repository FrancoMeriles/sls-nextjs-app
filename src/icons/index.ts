import HeartFull from './HeartFull'
import HeartOutline from './HeartOutline'

export { HeartFull, HeartOutline }
