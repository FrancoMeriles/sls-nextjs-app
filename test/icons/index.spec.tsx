import renderer from 'react-test-renderer'

import * as Icons from '@base/icons'

describe('Icons', () => {
  it('should render and match snapshot with required props', () => {
    const icons = Object.keys(Icons)
    icons.forEach((icon) => {
      const Icon = Icons[icon]
      const tree = renderer.create(<Icon />).toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
