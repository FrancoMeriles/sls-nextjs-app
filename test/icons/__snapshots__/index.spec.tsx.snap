// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icons should render and match snapshot with required props 1`] = `
<svg
  className="chakra-icon css-onkibi"
  focusable={false}
  viewBox="0 -28 512 512"
>
  <path
    d="M471.383 44.578C444.879 15.832 408.512 0 368.973 0c-29.555 0-56.621 9.344-80.45 27.77C276.5 37.07 265.605 48.45 256 61.73c-9.602-13.277-20.5-24.66-32.527-33.96C199.648 9.344 172.582 0 143.027 0c-39.539 0-75.91 15.832-102.414 44.578C14.426 72.988 0 111.801 0 153.871c0 43.3 16.137 82.938 50.781 124.742 30.992 37.395 75.535 75.356 127.117 119.313 17.614 15.012 37.579 32.027 58.309 50.152A30.023 30.023 0 00256 455.516c7.285 0 14.316-2.641 19.785-7.43 20.73-18.129 40.707-35.152 58.328-50.172 51.575-43.95 96.117-81.906 127.11-119.305C495.867 236.81 512 197.172 512 153.867c0-42.066-14.426-80.879-40.617-109.289zm0 0"
    fill="currentColor"
  />
</svg>
`;

exports[`Icons should render and match snapshot with required props 2`] = `
<svg
  className="chakra-icon css-onkibi"
  focusable={false}
  viewBox="0 -28 512 512"
>
  <path
    d="M256 455.516c-7.29 0-14.316-2.641-19.793-7.438-20.684-18.086-40.625-35.082-58.219-50.074l-.09-.078c-51.582-43.957-96.125-81.918-127.117-119.313C16.137 236.81 0 197.172 0 153.871c0-42.07 14.426-80.883 40.617-109.293C67.121 15.832 103.488 0 143.031 0c29.555 0 56.621 9.344 80.446 27.77C235.5 37.07 246.398 48.453 256 61.73c9.605-13.277 20.5-24.66 32.527-33.96C312.352 9.344 339.418 0 368.973 0c39.539 0 75.91 15.832 102.414 44.578C497.578 72.988 512 111.801 512 153.871c0 43.3-16.133 82.938-50.777 124.738-30.993 37.399-75.532 75.356-127.106 119.309-17.625 15.016-37.597 32.039-58.328 50.168a30.046 30.046 0 01-19.789 7.43zM143.031 29.992c-31.066 0-59.605 12.399-80.367 34.914-21.07 22.856-32.676 54.45-32.676 88.965 0 36.418 13.535 68.988 43.883 105.606 29.332 35.394 72.961 72.574 123.477 115.625l.093.078c17.66 15.05 37.68 32.113 58.516 50.332 20.961-18.254 41.012-35.344 58.707-50.418 50.512-43.051 94.137-80.223 123.469-115.617 30.344-36.618 43.879-69.188 43.879-105.606 0-34.516-11.606-66.11-32.676-88.965-20.758-22.515-49.3-34.914-80.363-34.914-22.758 0-43.653 7.235-62.102 21.5-16.441 12.719-27.894 28.797-34.61 40.047-3.452 5.785-9.53 9.238-16.261 9.238s-12.809-3.453-16.262-9.238c-6.71-11.25-18.164-27.328-34.61-40.047-18.448-14.265-39.343-21.5-62.097-21.5zm0 0"
    fill="currentColor"
  />
</svg>
`;

exports[`Icons should render and match snapshot with required props 3`] = `null`;

exports[`Icons should render and match snapshot with required props 4`] = `
<svg
  className="chakra-icon css-onkibi"
  focusable={false}
  viewBox="0 0 14 12"
>
  <path
    d="M2.474 7.2h7.225a2.7 2.7 0 1 0 0-5.4H7V.6h2.7a3.9 3.9 0 1 1 0 7.8H2.473l2.45 2.389-.839.859L.14 7.8l3.945-3.848.838.859L2.473 7.2z"
    fill="currentColor"
  />
</svg>
`;

exports[`Icons should render and match snapshot with required props 5`] = `
<svg
  className="chakra-icon css-onkibi"
  focusable={false}
  viewBox="0 0 18 15"
>
  <path
    d="M7.763 12.207a2.398 2.398 0 0 1-4.726 0H1.8a1.8 1.8 0 0 1-1.8-1.8V2.195a1.8 1.8 0 0 1 1.8-1.8h8.445a1.8 1.8 0 0 1 1.8 1.8v.568l3.322.035L18 6.821v5.386h-2.394a2.398 2.398 0 0 1-4.727 0H7.763zm-.1-1.2h3.182V2.195a.6.6 0 0 0-.6-.6H1.8a.6.6 0 0 0-.6.6v8.212a.6.6 0 0 0 .6.6h1.337a2.399 2.399 0 0 1 4.526 0zm7.843 0H16.8V7.179l-2.086-3.187-2.669-.029v5.76a2.399 2.399 0 0 1 3.461 1.284zm-2.263 1.99a1.198 1.198 0 1 0 0-2.395 1.198 1.198 0 0 0 0 2.396zm-7.843 0a1.198 1.198 0 1 0 0-2.395 1.198 1.198 0 0 0 0 2.396z"
    fill="currentColor"
    fillRule="nonzero"
  />
</svg>
`;

exports[`Icons should render and match snapshot with required props 6`] = `
<svg
  className="chakra-icon css-onkibi"
  focusable={false}
  viewBox="0 0 16 16"
>
  <g
    fill="currentColor"
    fillOpacity=".8"
    fillRule="evenodd"
  >
    <g
      fill="#000"
      fillRule="nonzero"
    >
      <path
        d="M4.996 1.4l-.001 1h6.067v-1h1.2v1h2.337v12.2H1.4V2.4h2.395l.001-1h1.2zM13.399 6.8H2.6v6.6h10.799V6.8zm-9.604-3.2L2.6 3.602v1.998h10.799V3.6L12.262 3.6v1h-1.2v-1H4.995l.001 1h-1.2l-.001-1z"
        transform="translate(-1128 -464) translate(1108 158) translate(20 254) translate(0 52)"
      />
    </g>
  </g>
</svg>
`;

exports[`Icons should render and match snapshot with required props 7`] = `
<svg
  className="chakra-icon css-onkibi"
  focusable={false}
  viewBox="0 0 20 20"
>
  <path
    d="M10 1.002a9 9 0 1 1 0 18 9 9 0 0 1 0-18zm0 1.2a7.8 7.8 0 1 0 0 15.6 7.8 7.8 0 0 0 0-15.6zm.6 2.376l.001 1.323c.972.109 1.774.484 2.393 1.083l-.804 1.044c-.624-.624-1.476-.912-2.316-.912-.828 0-1.356.396-1.356.996 0 .624.816.828 1.764 1.068 1.32.336 2.892.744 2.892 2.484 0 1.219-.784 2.277-2.573 2.448L10.6 15.44H9.4v-1.343c-1.131-.133-1.98-.598-2.574-1.233l.78-1.08a3.453 3.453 0 0 0 2.532 1.104c1.152 0 1.608-.564 1.608-1.092 0-.72-.852-.936-1.824-1.188l-.294-.073c-1.233-.315-2.55-.76-2.55-2.327 0-1.165.906-2.096 2.323-2.299L9.4 4.578h1.2z"
    fill="currentColor"
    fillOpacity=".8"
    fillRule="nonzero"
  />
</svg>
`;

exports[`Icons should render and match snapshot with required props 8`] = `
<svg
  className="chakra-icon css-onkibi"
  focusable={false}
  viewBox="0 0 14 19"
>
  <path
    clipRule="evenodd"
    d="M13.3471 6.34707C13.3471 9.39028 11.3722 13.1849 7.45665 17.7792L7 18.315L6.54334 17.7792C2.62775 13.1849 0.652924 9.39028 0.652924 6.34707C0.652924 2.84168 3.4946 0 7 0C10.5054 0 13.3471 2.84168 13.3471 6.34707ZM9.41002 6.595C9.41002 7.92602 8.33102 9.00503 7 9.00503C5.66898 9.00503 4.58997 7.92602 4.58997 6.595C4.58997 5.26398 5.66898 4.18497 7 4.18497C8.33102 4.18497 9.41002 5.26398 9.41002 6.595Z"
    fill="currentColor"
    fillRule="evenodd"
  />
</svg>
`;

exports[`Icons should render and match snapshot with required props 9`] = `
<svg
  className="chakra-icon css-onkibi"
  focusable={false}
  viewBox="0 0 14 15"
>
  <path
    d="M.399 3.89v-.6h.6c1.87 0 3.76-.783 5.68-2.374l.383-.317.383.317c1.92 1.59 3.81 2.374 5.68 2.374h.6v.6c0 5.633-2.165 9.242-6.473 10.679l-.19.063-.19-.063C2.564 13.132.4 9.523.4 3.89zm6.663-1.743c-1.8 1.4-3.62 2.179-5.455 2.32.135 4.725 1.947 7.648 5.455 8.898 3.508-1.25 5.32-4.173 5.455-8.898-1.835-.141-3.656-.92-5.455-2.32zm-.905 6.477l3.191-3.582.896.798-4.02 4.513-2.472-2.377.831-.865 1.574 1.513z"
    fill="currentColor"
    fillRule="nonzero"
  />
</svg>
`;

exports[`Icons should render and match snapshot with required props 10`] = `
<svg
  className="chakra-icon css-onkibi"
  focusable={false}
  viewBox="0 0 12 14"
>
  <path
    d="M5.305 10.354l-1.686 3.34-1.24-2.458H0l1.64-3.202a5.2 5.2 0 1 1 8.588.194l1.54 3.008H9.39l-1.241 2.458-1.673-3.315a5.266 5.266 0 0 1-1.171-.025zm-1.001-.237a5.201 5.201 0 0 1-1.975-1.234l-.693 1.353h1.357l.626 1.239.685-1.358zm3.192.064l.653 1.294.625-1.239h1.358l-.617-1.204a5.193 5.193 0 0 1-2.019 1.15zM6 9.2a4 4 0 1 0 0-8 4 4 0 0 0 0 8z"
    fill="currentColor"
    fillRule="nonzero"
  />
</svg>
`;

exports[`Icons should render and match snapshot with required props 11`] = `
<svg
  className="chakra-icon css-onkibi"
  focusable={false}
  viewBox="0 0 32 32"
>
  <path
    d="M16 8.85c1.593 0 3.064.521 4.253 1.402l-.962.91A5.851 5.851 0 1016 21.85a5.851 5.851 0 005.772-6.808l1.096-1.038A7.15 7.15 0 1116 8.85zm5.642 1.89l1.028 1.028-6.69 6.703-3.18-3.18 1.029-1.03 2.152 2.153 5.66-5.675z"
    fill="currentColor"
    fillRule="nonzero"
  />
</svg>
`;

exports[`Icons should render and match snapshot with required props 12`] = `
<svg
  className="chakra-icon css-onkibi"
  focusable={false}
  viewBox="0 0 17 17"
>
  <path
    d="M6.15 5.464l.86-.334L9.576 0l1.017.509a3 3 0 0 1 1.473 3.719l-.286.777h2.8a2.4 2.4 0 0 1 2.243 3.256l-1.205 3.159a4.2 4.2 0 0 1-4.682 2.635l-3.546-.65H6.15v1.2H2v-10.2h4.15v1.06zm0 1.288v5.453h1.294l.108.01 3.6.66a3 3 0 0 0 3.343-1.883L15.7 7.833a1.2 1.2 0 0 0-1.12-1.628h-4.52l.88-2.392a1.8 1.8 0 0 0-.829-2.202l-2.234 4.47-1.728.67zM3.2 13.404h1.75v-7.8H3.2v7.8z"
    fill="currentColor"
    fillRule="nonzero"
  />
</svg>
`;
